wallets/rn_cli_wallet_068_5/src/screens/HomeScreen.tsx

```diff
@@ -0,0 +1,289 @@
+    flex: 1,
+    padding: 16,
+    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
+  };
+
+  async function handleAccept() {
+    const {id, params} = pairedProposal;
+    const {requiredNamespaces, relays} = params;
+
+    if (pairedProposal) {
+      const namespaces: SessionTypes.Namespaces = {};
+      Object.keys(requiredNamespaces).forEach(key => {
+        const accounts: string[] = [];
+        requiredNamespaces[key].chains.map(chain => {
+          [currentETHAddress].map(acc => accounts.push(`${chain}:${acc}`));
+        });
+
+        namespaces[key] = {
+          accounts,
+          methods: requiredNamespaces[key].methods,
+          events: requiredNamespaces[key].events,
+        };
+      });
+
+      await web3wallet.approveSession({
+        id,
+        relayProtocol: relays[0].protocol,
+        namespaces,
+      });
+
+      setApprovalModal(false);
+      setSuccessPair(true);
+    }
+  }
+
+  const handleCancel = () => {
+    setCopyDialog(false);
+  };
+
+  async function pair() {
+    const pairing = await _pair({uri: WCURI});
+    setCopyDialog(false);
+    setWCUri('');
+    if (Platform.OS === 'android') {
+      setApprovalModal(true);
+    }
+    return pairing;
+  }
+
+  // ToDo / Consider: How best to move onSessionProposal() + onSessionRequest() + the if statement Listeners.
+  // Know there is an events config we did in web-examples app
+  const onSessionProposal = useCallback(
+    (proposal: SignClientTypes.EventArguments['session_proposal']) => {
+      setPairedProposal(proposal);
+    },
+    [],
+  );
+
+  const onSessionRequest = useCallback(
+    async (requestEvent: SignClientTypes.EventArguments['session_request']) => {
+      const {topic, params} = requestEvent;
+      const {request} = params;
+      const requestSessionData =
+        web3wallet.engine.signClient.session.get(topic);
+
+      switch (request.method) {
+        case EIP155_SIGNING_METHODS.ETH_SIGN:
+        case EIP155_SIGNING_METHODS.PERSONAL_SIGN:
+          setRequestSession(requestSessionData);
+          setRequestEventData(requestEvent);
+          setSignModal(true);
+          return;
+
+        case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA:
+        case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA_V3:
+        case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA_V4:
+          setRequestSession(requestSessionData);
+          setRequestEventData(requestEvent);
+          setSignTypedDataModal(true);
+          return;
+        case EIP155_SIGNING_METHODS.ETH_SEND_TRANSACTION:
+        case EIP155_SIGNING_METHODS.ETH_SIGN_TRANSACTION:
+          setRequestSession(requestSessionData);
+          setRequestEventData(requestEvent);
+          setSendTransactionModal(true);
+          return;
+      }
+    },
+    [],
+  );
+
+  useEffect(() => {
+    if (web3wallet) {
+      web3wallet.on('session_proposal', onSessionProposal);
+      web3wallet.on('session_request', onSessionRequest);
+    }
+  }, [
```

Comment:
why do the listeners have to be subscribed on each value change?
also consider adding `removeEventListener` with max listeners very fast
```
return () => {
 web3wallet.removeEventListener('session_proposal', onSessionProposal);
  web3wallet.removeEventListener('session_request', onSessionRequest);
}

Reply:
Switched it to just popup on modal state.
```
 if (approvalModal || copyDialog || signModal || sendTransactionModal) {
      web3wallet.on('session_proposal', onSessionProposal);
      web3wallet.on('session_request', onSessionRequest);
    }
    ```
    
Will refactor later similar to web-examples on revisit